addon/components/flash-message.js

```diff
@@ -2,127 +2,110 @@ import { htmlSafe, classify } from '@ember/string';
 import Component from '@ember/component';
 import { isPresent } from '@ember/utils';
 import { run } from '@ember/runloop';
-import { computed, set, get } from '@ember/object';
+import { action, computed, set } from '@ember/object';
+import { and, bool, readOnly, not } from '@ember/object/computed';
+import { tagName } from '@ember-decorators/component';
 import layout from '../templates/components/flash-message';
-import getWithDefault from '../utils/get-with-default';
-
-const {
-  and,
-  bool,
-  readOnly,
-  not
-} = computed;
-const {
-  next,
-  cancel
-} = run;
-
-export default Component.extend({
-  layout,
-  active: false,
-  messageStyle: 'bootstrap',
-  classNames: ['flash-message'],
-  classNameBindings: ['alertType', 'active', 'exiting'],
-  attributeBindings: ['aria-label', 'aria-describedby', 'role'],
-
-  showProgress: readOnly('flash.showProgress'),
-  notExiting: not('exiting'),
-  showProgressBar: and('showProgress', 'notExiting'),
-  exiting: readOnly('flash.exiting'),
-  hasBlock: bool('template').readOnly(),
-
-  alertType: computed('flash.type', {
-    get() {
-      const flashType = getWithDefault(this, 'flash.type', '');
-      const messageStyle = getWithDefault(this, 'messageStyle', '');
-      let prefix = 'alert alert-';
-
-      if (messageStyle === 'foundation') {
-        prefix = 'alert-box ';
-      }
-
-      return `${prefix}${flashType}`;
-    }
-  }),
 
-  flashType: computed('flash.type', {
-    get() {
-      const flashType = getWithDefault(this, 'flash.type', '');
+const { next, cancel } = run;
 
-      return classify(flashType);
-    }
-  }),
+@tagName('')
+export default class FlashMessage extends Component {
+  layout = layout;
+  active = false;
+  messageStyle = 'bootstrap';
 
-  didInsertElement() {
-    this._super(...arguments);
-    const pendingSet = next(this, () => {
-      set(this, 'active', true);
-    });
-    set(this, 'pendingSet', pendingSet);
-    this.set('_mouseEnterHandler', this._mouseEnter.bind(this));
-    this.set('_mouseLeaveHandler', this._mouseLeave.bind(this));
-    this.element.addEventListener('mouseenter', this._mouseEnterHandler);
-    this.element.addEventListener('mouseleave', this._mouseLeaveHandler);
-  },
-
-  willDestroyElement() {
-    this._super(...arguments);
-    this.element.removeEventListener('mouseenter', this._mouseEnterHandler);
-    this.element.removeEventListener('mouseleave', this._mouseLeaveHandler);
-  },
-
-  progressDuration: computed('flash.showProgress', {
-    get() {
-      if (!get(this, 'flash.showProgress')) {
-        return false;
-      }
-
-      const duration = getWithDefault(this, 'flash.timeout', 0);
-
-      return htmlSafe(`transition-duration: ${duration}ms`);
+  @readOnly('flash.showProgress')
+  showProgress;
+
+  @not('exiting')
+  notExiting;
+
+  @and('showProgress', 'notExiting')
+  showProgressBar;
+
+  @readOnly('flash.exiting')
+  exiting;
+
+  @bool('template')
+  hasBlock;
+
+  @computed('flash.type')
```

Comment:
Missing a `messageStyle` dependency? Newer versions of eslint-plugin-ember can find and fix these kinds of errors.

Reply:
Not sure what you mean? I don't see the issue?

Reply:
`alertType` won't update when `this.messageStyle` changes. Just needs line 33 updated to `@computed('flash.type', 'messageStyle')`

Reply:
Oh gotcha, I didn't look at the body heh. I'll update computed and see if I can get the linter to pick up this issue.